src/coverlet.core/Coverage.cs

```diff
@@ -58,20 +59,22 @@ public string Identifier
             _mergeWith = mergeWith;
             _useSourceLink = useSourceLink;
             _logger = logger;
+            _fileSystem = fileSystem;
             _instrumentationHelper = instrumentationHelper;
 
             _identifier = Guid.NewGuid().ToString();
             _results = new List<InstrumenterResult>();
         }
 
-        public Coverage(CoveragePrepareResult prepareResult, ILogger logger, IInstrumentationHelper instrumentationHelper)
+        public Coverage(CoveragePrepareResult prepareResult, ILogger logger, IFileSystem fileSystem, IInstrumentationHelper instrumentationHelper)
```

Comment:
please keep `IFileSystem` as last parameter we'll add new service at the end of the signature as a rule of thumb

Reply:
Done